packages/image_picker/image_picker_android/android/src/main/java/io/flutter/plugins/imagepicker/FileUtils.java

```diff
@@ -25,55 +25,46 @@
 
 import android.content.ContentResolver;
 import android.content.Context;
+import android.database.Cursor;
 import android.net.Uri;
+import android.provider.MediaStore;
 import android.webkit.MimeTypeMap;
-import java.io.File;
-import java.io.FileOutputStream;
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.OutputStream;
+import java.io.*;
+import java.util.UUID;
 
 class FileUtils {
 
+  @SuppressWarnings("IOStreamConstructor")
   String getPathFromUri(final Context context, final Uri uri) {
-    File file = null;
-    InputStream inputStream = null;
-    OutputStream outputStream = null;
-    boolean success = false;
-    try {
-      String extension = getImageExtension(context, uri);
-      inputStream = context.getContentResolver().openInputStream(uri);
-      file = File.createTempFile("image_picker", extension, context.getCacheDir());
-      file.deleteOnExit();
-      outputStream = new FileOutputStream(file);
-      if (inputStream != null) {
-        copy(inputStream, outputStream);
-        success = true;
-      }
-    } catch (IOException ignored) {
-    } finally {
-      try {
-        if (inputStream != null) inputStream.close();
-      } catch (IOException ignored) {
+    try (InputStream inputStream = context.getContentResolver().openInputStream(uri)) {
```

Comment:
For increased clarity in the future, document what the intended file naming convention is and any fallbacks